Python/ast_opt.c

```diff
@@ -268,38 +263,44 @@ fold_subscr(expr_ty node, PyArena *arena)
     return make_const(node, newval, arena);
 }
 
+static int
+fold_iter(expr_ty arg, PyArena *arena)
+{
+    PyObject *newval;
+    if (arg->kind == List_kind) {
+        newval = make_const_tuple(arg->v.List.elts);
+    }
+    else if (arg->kind == Set_kind) {
+        newval = make_const_tuple(arg->v.Set.elts);
+        if (newval) {
+            Py_SETREF(newval, PyFrozenSet_New(newval));
+        }
+    }
+    else {
+        return 1;
+    }
+    return make_const(arg, newval, arena);
+}
+
 static int
 fold_compare(expr_ty node, PyArena *arena)
```

Comment:
Ditto here.

Reply:
This already is documented inside the function. In future the optimization for literal operands (like `1 < 2`) can be added here, but currently it optimizes only the particular case of 'in' and 'not in' operators.